app/lib/sequel/plugins/time_machine.rb

```diff
@@ -43,6 +43,24 @@ def point_in_time
         def relevant_query?
           Thread.current[::TimeMachine::THREAD_RELEVANT_KEY]
         end
+
+        def validity_dates_filter(table = self,
+                                  start_column: :validity_start_date,
+                                  end_column: :validity_end_date)
+          return Sequel.expr(true) if point_in_time.blank?
+
+          table_name = if table.is_a?(Class) && table < Sequel::Model
```

Comment:
Yeh, I had to look this up - its a bit quirky because I'm testing the model class rather then an a model instance

Reply:
Thinking about this, to implement the plugin you'd need to inherit from the Sequel::Model ancestor. Is there any point in checking this?

Reply:
I mean it doesn't hurt

Reply:
Yeh - but table is a value passed in - its intended to handle different scenarios - if its given a model class, it'll work with that and fetch out `table_name` but if its given a symbol for the table then it needs to treat it as a table name and not be clever about it

I ended up adding that stuff because the existing methods aren't particularly flexible and I needed something I could use in a few different scenarios in the complex joins